lib/mongo/client.rb

```diff
@@ -679,23 +682,23 @@ def update_options(new_options)
           options.delete(:write_concern)
         end
 
-        # If the user specifies new auto_encryption_options, teardown the
-        # existing encryption infrastructure
-        if opts.key?(:auto_encryption_options)
-          should_close_encrypter = true
-        end
-
-        # If the new auto_encryption_options are not nil, then set up the
-        # encrypter again with the new options
-        if opts[:auto_encryption_options]
-          should_set_new_encryption_options = true
-        end
-
         options.update(opts)
         @options = options.freeze
 
-        close_encrypter if should_close_encrypter
-        build_encrypter if should_set_new_encryption_options
+        auto_encryption_options_changed =
+          @options[:auto_encryption_options] != old_options[:auto_encryption_options]
```

Comment:
I'm not sure how people feel about multi-line variable assignment. Open to other styles here :)

Reply:
This looks good :+1: